app/src/main/java/com/pvp/app/ui/screen/friends/FriendsViewModel.kt

```diff
@@ -180,6 +194,55 @@ class FriendsViewModel @Inject constructor(
             )
         }
     }
+
+    fun getMutualFriends(friendEmail: String) {
+        viewModelScope.launch {
+            val friendObject = friendService
+                .get(friendEmail)
+                .firstOrNull()
+
+            val mutualFriendsList =
+                friendObject?.friends?.map { it.email }
+                    ?.intersect(userFriends.value.map { it.user.email }
+                        .toSet())
+
+            if (mutualFriendsList != null) {
+                val mutualFriendsUsers = mutualFriendsList.mapNotNull {
+                    val user = userService.get(it).firstOrNull()
+
+                    user?.let {
+                        FriendEntry(
+                            avatar = decorationService.getAvatar(user),
```

Comment:
This is not using flow (your added decoration method). So if any changes are being made to the decorations, they will not trigger update on this avatar.

It is not ideal, but it is also only shown when user info is clicked, right? So, since it is not a place that is seen 24/7, this should be fine as it is, since on each info open it should recompute.

_Also, decorations being updated is hardly going to happen, but it can._